test/unit/errors.test.js

```diff
@@ -2,10 +2,36 @@
 
 const expect = require('chai').expect;
 const { getSymbolFrom } = require('../tools/utils');
-const MongoNetworkError = require('../../src/error').MongoNetworkError;
+const { MongoNetworkError } = require('../../src/index');
+const {
+  PoolClosedError: MongoPoolClosedError,
+  WaitQueueTimeoutError: MongoWaitQueueTimeoutError
+} = require('../../src/cmap/errors');
 
-describe('MongoErrors', function () {
-  describe('MongoNetworkError', function () {
+describe('MongoErrors', () => {
+  // import errors as object
+  let errorClasses = Object.fromEntries(
+    Object.entries(require('../../src/index')).filter(([key]) => key.endsWith('Error'))
+  );
+  errorClasses = { ...errorClasses, MongoPoolClosedError, MongoWaitQueueTimeoutError };
+
+  for (const errorName in errorClasses) {
+    describe(errorName, () => {
+      it(`name should be read-only`, () => {
+        // Dynamically create error class with message
+        let error = new errorClasses[errorName]('generated by test');
+        // expect name property to be class name
+        expect(error).to.have.property('name', errorName);
+
+        try {
+          error.name = 'renamed by test';
+          expect(error).to.have.property('name', 'generated by test');
```

Comment:
this expect should be outside the try/catch block since you don't want to swallow the error thrown by the expect in the case of failure, otherwise looks good :)

Reply:
Oops, sorry! Just fixed that and replaced `'generated by test'` with `errorName` because it was also a typo.